plugins/optimization-detective/storage/class-od-rest-url-metrics-store-controller.php

```diff
@@ -0,0 +1,359 @@
+<?php
+/**
+ * REST API integration for the plugin: OD_REST_URL_Metrics_Store_Controller.
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * OD_REST_URL_Metrics_Store_Controller class
+ *
+ * @since n.e.x.t
+ *
+ * @access private
+ */
+class OD_REST_URL_Metrics_Store_Controller {
+
+	/**
+	 * Get the namespace for optimization-detective
+	 *
+	 * @since n.e.x.t
+	 * @return non-empty-string Namespace.
+	 */
+	public static function get_namespace(): string {
+		return OD_REST_API_NAMESPACE; // @phpstan-ignore constant.deprecated (To be replaced with string literal when constant is removed.)
+	}
+
+	/**
+	 * Gets the route for storing a URL Metric.
+	 *
+	 * Note the `:store` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way
+	 * that does not strictly follow the standard usage. Namely, submitting a POST request to this endpoint will either
+	 * create a new `od_url_metrics` post, or it will update an existing post if one already exists for the provided slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @link https://google.aip.dev/136
+	 *
+	 * @return non-empty-string Route.
+	 */
+	public static function get_route(): string {
+		return OD_URL_METRICS_ROUTE; // @phpstan-ignore constant.deprecated (To be replaced with string literal when constant is removed.)
+	}
+
+	/**
+	 * Registers endpoint for storage of URL Metric.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @see od_compose_site_health_result()
+	 */
+	public function register_endpoint(): void {
```

Comment:
See above, per https://github.com/WordPress/performance/pull/1865#issuecomment-2701757708 this method should not be included in the class. Instead the class should have a method (e.g. `get_registration_args()`) to return the data for the third `register_rest_route` parameter as an array.